code/segmentedtickerplant/stplog.q

```diff
@@ -190,55 +198,61 @@ endofdaydata:{
 
 // Send close of period message to subscribers, update logging period times
 // roll logs if flag is specified - we don't want to roll logs if end-of-day is also going to be triggered
-$[.sctp.chainedtp;
-  endofperiod:{[x;y;z]
+$[not null .sctp.sctploggingmode;
```

Comment:
For end of day / end of period... instead of creating the definition like this, can you just use if statements within the function definition, rather than a switch where we declare the function. 
The way you have it now is what we want to do for the functions that are called often (e.g. every update) but it makes the code more confusing. So for the functions that are called rarely (endofday/endofperiod) can you just put if statements  within the function definition, rather than having two separate definitions